meeting.js

```diff
@@ -0,0 +1,330 @@
+    console.error('Error getting local streams:', error);
+    throw error;
+  }
+}
+
+// 9) Joining the Meeting and add Local Streams to meeting.
+async function joinMeetingWithMedia(localStreams) {
+  try {
+    const meetingOptions = {
+      mediaOptions: {
+        allowMediaInLobby: true,
+        shareAudioEnabled: false,
+        shareVideoEnabled: false,
+        localStreams,      
+      },
+    };
+
+    await createdMeeting.joinWithMedia(meetingOptions);
+
+    document.getElementById('joinMeetingModal').style.display = 'block';
+    joinMeetingButton.style.display = 'none';
+  } catch (error) {
+    console.error('Error joining meeting with media:', error);
+    throw error;
+  }
+}
+
+// 10) Join Meeting Listener and Calling Above Functions(6 to 9).
+joinMeetingButton.addEventListener('click', joinMeeting);
+async function joinMeeting() {
+  document.getElementById("joinMeetingButton").style.display = "none";
+  document.getElementById("localvideoimage").style.display = "none";
+  $('#joinMeetingModal').modal('show');
+  var element = document.querySelector('.container3');
+  element.classList.add('blur');
+  try {
+    await createMeeting();
+    setMediaListeners();
+    localStream = await getLocalStreams();
+    createdMeeting.setRemoteQualityLevel('HIGH');
+    await joinMeetingWithMedia(localStream);
+  } catch (error) {
+    console.error('Error joining meeting:', error);
+  }
+}
+
+// 11) Leaving Meeting Listener and Function.
+leaveMeetingButton.addEventListener('click', leaveMeeting);
+async function leaveMeeting() {
+  try {
+    const confirmLeave = window.confirm("Are you sure you want to leave the meeting?");
+    if(confirmLeave){
+      await createdMeeting.leave();
+      joinMeetingButton.style.display = 'block';
+      $('#joinMeetingModal').modal('hide');
+      videoButton.innerHTML = '<i class="fas fa-video icon green"></i>';
+      microphoneButton.innerHTML = '<i class="fas fa-microphone icon green"></i>';
+      microphoneButton.style.padding = "14px 21px";
+      bnrButton.classList.remove('selected');
+      vbgButton.classList.remove('selected');
+      createdMeeting=null;
+      localStream = null;
+      isMuted = true;
+      isVideoStarted = true;
+      vbgEffect = false;
+      
+    }
+  } catch (error) {
+    console.error('Error leaving meeting:', error);
+    $('#joinMeetingModal').modal('hide');
+
+  }
+}
+
+// 12) Start and Stop Button using toggleVideo Function.
+videoButton.addEventListener('click', toggleVideo);
+async function toggleVideo() {
+  if (!localStream || !localStream.camera || !localStream.camera.outputStream) {
+    console.error('No local video stream available.');
+    return;
+  }
+  // Stop Video
+  if (isVideoStarted) {
+    await createdMeeting.unpublishStreams([localStream.camera]);
+    localVideo.srcObject = null;
+    document.getElementById("localvideoimage").style.display = "block";
+    videoButton.innerHTML = '<i class="fas fa-video-slash icon red"></i>';

```

Comment:
Instead of replacing the whole HTML, we should simply be replacing the class name

Reply:
this is seperate icon to mute and unmute so if we give the class name directly the then it will not replicate the icon switches like we are using different icons for mute,unmute start,stop .So switching from one icon to another icon we use innerHtml rather than giving class name

Reply:
Like we have to remove existing class and add the related class so i had updated with the innerHTML

Reply:
Have you explored the [javascript classList](https://www.w3schools.com/jsref/prop_element_classlist.asp) property?

Reply:
Yes I had Explored Class List Property And applied But it did not Works As Expected Compare to jQuery

Reply:
I had applied the through class list it did not expect from the result as innerHTML.And why we use class list instead of innerHtml for the code.

Reply:
Had a long 1:1 conversation on why.

innerHTML removes the whole element and pushes new element on the UI causing a re-render while classList only switches the CSS rules that are already loaded and parsed in the DOM. That is why it is good to prefer classList than switching HTML